### DISCUSSION TITLE
error importing extensions w/ sveltekit

### DISCUSSION THREAD

**Discussion by rawwerks** [donmccurdy/glTF-Transform#1224] [Q&A] [answered]
i'm running into this console error when trying to import compression via: `import { EXTMeshoptCompression } from '@gltf-transform/extensions';` (trying to follow this example: https://gltf-transform.dev/modules/extensions/classes/EXTMeshoptCompression ) 

`SyntaxError: ambiguous indirect export: EXTMeshoptCompression`

it's a sveltekit/vite app, and it runs `@gltf-transform/core` and `@gltf-transform/functions` just fine, and the app builds fine until it hits that import, which then gives a 500 with the syntax error above in the console. 

i've got this installed:
```
sveltekit-app@0.0.1 /my/path
├── @gltf-transform/extensions@3.10.0
└─┬ @gltf-transform/functions@3.10.0
  └── @gltf-transform/extensions@3.10.0 deduped
```

```
sveltekit-app@0.0.1 /my/path
├── @gltf-transform/core@3.10.0
├─┬ @gltf-transform/extensions@3.10.0
│ └── @gltf-transform/core@3.10.0 deduped
└─┬ @gltf-transform/functions@3.10.0
  └── @gltf-transform/core@3.10.0 deduped
```

i'm using yarn to install everything, maybe that's the problem?

in case it's useful information, there is no problem with this import: 
`import { MeshoptDecoder, MeshoptEncoder } from 'meshoptimizer';`

and also, if i uncomment the problematic import (first sentence of this post) while the app is running, i get this modified console error:

```

SyntaxError: ambiguous indirect export: EXTMeshoptCompression [client.ts:115:12](http://localhost:5173/@vite/client.ts)
[hmr] Failed to reload /src/routes/+page.svelte. This could be due to syntax errors or importing non-existent modules. (see errors above)

```

for fun, i tried seeing if the issue is with any extension:

`import { EXTMeshGPUInstancing } from '@gltf-transform/extensions';` causes the same console error, so i think it has to do with not installing extensions correctly, not specifically the meshoptimizer extension.

**Comment by donmccurdy** [comment]
Hi! I'll probably need a simple way to reproduce the error to investigate. I went ahead and did a basic `npm create svelte@latest my-app` and added the following imports to `+page.svelte`...

```js
import { EXTMeshoptCompression } from '@gltf-transform/extensions';
console.log(EXTMeshoptCompression);
```

... but couldn't see any errors after running `yarn dev`.

I'll venture a guess this has something to do with ESM vs CJS modules, and how SvelteKit compiles server-side and/or client-side code depending on how your configuration. If your application isn't ESM-only, or compiles to CJS for some reason, it might be worth testing with just ESM. glTF Transform does provide both ESM and CJS exports.

**Comment by rawwerks** [comment, marked as answer]
so i'm not sure exactly what fixed it, because this error caused me to basically upgrade everything, but my best guess is that it has something to do with using the vitePreProcess, which i think was unnecessary anyways because i already have svelte preprocess. once i got rid of this, it started loading.

```
import { vitePreprocess } from '@sveltejs/kit/vite';
```

but now that it's loading i'm sure i'll have more questions to get it working;)
